web/docs/language/features.md

```diff
@@ -178,138 +178,249 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
+Here's an implementation of a simple Action:
 
-In practice, this means that without really even thinking about it, Wasp will make sure to keep the queries up to date for you in regard with the changes done by actions.
+```js title=actions.js
+export const sayHi = async () => {
+  console.log('The client said Hi!')
+}
+```
+Its corresponding declaration in Wasp:
 
-On the other hand, this kind of automatic invalidation of cache can be wasteful (updating when not needed) and will not work if other resources than entities are used. In that case, make sure to use mechanisms provided by react-query for now, and expect more direct support in Wasp for handling those use cases in a nice, elegant way.
+```c title="main.wasp"
+// ...
 
+action sayHi {
+  fn: import { sayHi } from "@ext/actions.js"
+}
+```
+And an example of how to import and call the declared Action:
 
-### Action
+```js
+import sayHi from '@wasp/actions/sayHi'
+
+// ...
+
+sayHi()
+```
+
+
+More differences and Action/Query specific features will come in future versions of Wasp.
+
+### Cache Invalidation
+One of the trickiest parts of managing a web app's state is making sure the data shown by the queries are up to date.
+Since Wasp uses _react-query_ for Query management, we must make sure to invalidate its caches whenever they become stale.
+
+It's possible to invalidate the caches manually through several mechanisms _react-query_ provides (e.g., refetch, direct invalidation).
+However, since manual cache invalidation quickly becomes complex and error-prone, Wasp offers a quicker and a more effective solution to get you started: **automatic Entity-based Query cache invalidation**.
+Because Actions can (and most often do) modify the state while Queries read it, Wasp invalidates a Query's cache whenever an Action that uses the same Entity is executed.
+
+For example, let's assume that Action `A1` and Query `Q1` both use Entity `E1`. If `A1` is executed, `Q1`'s cached result may no longer be up-to-date.
+Wasp will therefore invalidate it, making `Q1` up to date again.
+
+In practice, this means that Wasp keeps the queries "fresh" without requiring you to think about cache invalidation.
 
-Actions are very similar to Queries, so similar that we will only list the differences:
-1. They can modify state (queries can't).
-2. There is no special React hook for them (like `useQuery` for Queries), you just call them directly.
-3. They are declared in Wasp in same way as Queries, but keyword is `action`, not `query`.
+On the other hand, such excessive cache invalidation can quickly become wasteful and will not work for any resource other than Entities.
+Wasp does not yet provide native support for such use cases, so please use _react-query_'s built-in mechanisms for the time being.
+You can expect Wasp to support a more elegant solution in the future.
```

Comment:
Maybe for this part I would stick with the original
```
On the other hand, this kind of automatic invalidation of cache can be wasteful (updating when not needed) and will not work if other resources than entities are used. In that case, make sure to use mechanisms provided by react-query for now, and expect more direct support in Wasp for handling those use cases in a nice, elegant way.
```

statement, as it is a bit more optimistic / exciting (`nice, elegant solution` vs `more elegant solution`).